src/NuGet.Clients/NuGet.Console/WpfConsole/WpfConsole.cs

```diff
@@ -840,7 +840,12 @@ public void SetConsoleWidth(int consoleWidth)
 
             public void SetExecutionMode(bool isExecuting)
             {
-                Invoke(() => _impl.SetExecutionMode(isExecuting));
+                Invoke(() =>
+                {
+                    ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
Is the analyzer complain about this? 

The `Invoke` method itself switch on the UI thread.

Reply:
Yes, the analyzer specifically complained about this.  I only addressed threading issues that the analyzer complained about.